addons/project/models/project.py

```diff
@@ -62,8 +64,29 @@
     'child_ids',
     'parent_id',
     'priority',
+    'state',
+    'state_approval_mode'
```

Comment:
if those fields are writable in project sharing then they are also readable. 🙂 
That is, no need to add those fields in the readable fields. 🙂 

https://github.com/odoo/odoo/blob/b9f0963ce3ee318a87aa011bc1f0c7b2a5013169/addons/project/models/project.py#L1347-L1349